### PR TITLE
feat(@angular-ru/common/utils): add downloadFile utility

### PR DESCRIPTION
**Pull request by nikitasadchikov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/common/utils/src/download-file.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    blob: Blob | File;
+    name: string;
+    extension: string;
+}
+
+export function downloadFile(file: FileToDownloadInfo): void {
+    const anchor: HTMLAnchorElement = document.createElement('a');
+    const url: string = window.URL.createObjectURL(file.blob);
+    anchor.href = url;
+    anchor.download = `${file.name}.${file.extension}`;
```
Comment (Schirbak) [inline packages/common/utils/src/download-file.ts:11]: @nikitasadchikov @splincode не находите это злом? а вдруг ~~нам~~ понадобится скачивать Dockerfile?

File: packages/ng-excel-builder/src/excel-builder.service.ts [hunk 2]
```diff
@@ -250,8 +242,8 @@ export class ExcelBuilderService {
                 return new Blob([UTF8, template], { type: 'application/vnd.ms-excel;charset=UTF-8' });
             }, workbook)
             .then((blob: Blob): void => {
-                const fileName: string = workbook.filename;
-                ExcelBuilderService.downloadFile(blob, fileName);
+                const name: string = `${workbook.filename}.${toFormatDateTime()}`;
```
Comment (Schirbak) [inline packages/ng-excel-builder/src/excel-builder.service.ts:245]: ставлю класс

File: packages/common/utils/src/download-file.ts (hunk 1 above)
Comment (splincode) [reply packages/common/utils/src/download-file.ts:11]: @Schirbak Yes, you are right, so I suggest doing

```ts
const filePath = [file.name, file.extension].filter(Boolean).join('.');
```

where `name` and `extension` is nullable value

and throw exception when name and extensions is empty

File: packages/common/utils/src/download-file.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+interface FileToDownloadInfo {
+    blob: Blob | File;
+    name: string;
```
Comment (splincode) [inline packages/common/utils/src/download-file.ts:3]: `name?: string | null;`
`extension?: string | null;`

File: packages/common/utils/src/download-file.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+interface FileToDownloadInfo {
```
Comment (splincode) [inline packages/common/utils/src/download-file.ts:1]: export

**Review by splincode — CHANGES_REQUESTED** [review]

File: packages/common/utils/src/download-file.ts (hunk 1 above)
Comment (Schirbak) [reply packages/common/utils/src/download-file.ts]: Zachem nullable? Zachem voobshe extension otdelno. Mozhno zhe prosto 
```ts
downloadFile(blob, `${myName}.xlsx`)

Comment (Schirbak) [reply packages/common/utils/src/download-file.ts]: ну так-то конечно типа вкусовщина, лан

**Review by Schirbak — DISMISSED** [review]

**Review by mansurishmuhametov — DISMISSED** [review]

**Review by splincode — APPROVED** [review]

**Review by Schirbak — APPROVED** [review]

File: packages/common/utils/src/download-file.ts (hunk 1 above)
Comment (splincode) [reply packages/common/utils/src/download-file.ts]: you can use overload for current util))

**Pull request merged**